### DISCUSSION TITLE
ERROR/MainProcess] consumer: Cannot connect to amqp://guest:**@127.0.0.1:5672//: [Errno 111] Connection refused. apache airflow celery worker

### DISCUSSION THREAD

**Discussion by Finrootz** [celery/celery#8303] [Q&A]
I deploy Airflow on aws so each component (web, scheduler, worker, and flower) is containerized and distributed among nodes. I set
{{AIRFLOW__CELERY__BROKER_URL}} in environment variables and it can be seen when I run {{printenv}} in a container, but when we are connecting a worker to broker (redis node) getting the below
exception
[2023-06-07 10:21:29,738: ERROR/MainProcess] consumer: Cannot connect to amqp://guest:**@127.0.0.1:5672//: [Errno 111] Connection refused

broker ulr looks like in airflow.cfg

broker_url = redis://$REDIS_HOST:$REDIS_PORT/1

please see the docker file for the component

FROM apache/airflow:2.6.1-python3.10
USER root
LABEL version="1.1"
LABEL maintainer="Finrootz"

ENV AIRFLOW_HOME=/usr/local/airflow

RUN apt-get update
&& apt-get install -y --no-install-recommends gcc libc6-dev libcurl4-openssl-dev libssl-dev \
&& apt-get autoremove -yqq --purge
&& apt-get clean
&& rm -rf /var/lib/apt/lists/*

COPY config/entrypoint.sh /entrypoint.sh
RUN chmod +x /entrypoint.sh
COPY config/airflow.cfg ${AIRFLOW_HOME}/airflow.cfg
COPY dags ${AIRFLOW_HOME}/dags

RUN chown -R airflow: ${AIRFLOW_HOME}

ENV PYTHONPATH ${AIRFLOW_HOME}

USER airflow

RUN pip install --upgrade pip
&& pip install "apache-airflow[celery,postgres,amazon,async,jdbc,pandas,redis]==2.6.1" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.6.1/constraints-3.7.txt"

COPY requirements.txt .
RUN pip install --user --no-cache-dir -r requirements.txt

EXPOSE 8080 5555 8793

WORKDIR ${AIRFLOW_HOME}
ENTRYPOINT ["/entrypoint.sh"]

**Comment by tharindu-s-rajapaksha** [comment]
**Check Your Broker Configuration**
```
CELERY_BROKER_URL = getattr(settings, "CELERY_BROKER_URL", settings.REDIS_URL)
CELERY_RESULT_BACKEND = getattr(settings, "CELERY_RESULT_BACKEND", settings.REDIS_URL)

print("CELERY_BROKER_URL:", CELERY_BROKER_URL)  # Check Your Broker Configuration
```
